apps/studio/src/server/modules/resource/resource.router.ts

```diff
@@ -535,4 +542,158 @@ export const resourceRouter = router({
 
       return ancestors.reverse().slice(0, -1)
     }),
+
+  search: protectedProcedure
+    .input(searchSchema)
+    .query(async ({ input: { siteId, query = "", cursor: offset, limit } }) => {
+      interface ResourceInterface {
+        id: string
+        title: string
+        type: string
+        parentId: string | null
+        lastUpdatedAt: Date | null
+        fullPermalink: string
+      }
```

Comment:
note - this has been extracted out into another file in https://github.com/opengovsg/isomer/pull/882 - will not duplicate the effort here